decidim-core/lib/decidim/content_parsers/resource_parser.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ContentParsers
+    # A parser that searches mentions of Resources in content.
+    #
+    # @see BaseParser Examples of how to use a content parser
+    class ResourceParser < BaseParser
+      # Matches a URL
+      URL_REGEX_SCHEME = '(?:http(s)?:\/\/)'
+      URL_REGEX_CONTENT = '[\w.-]+[\w\-\._~:\/?#\[\]@!\$&\'\(\)\*\+,;=.]+'
+      URL_REGEX_END_CHAR = '[\d]'
+      # Matches a mentioned resource ID (~(d)+ expression)
+      ID_REGEX = /~(\d+)/
+
+      # Replaces found mentions matching an existing
+      # Resource with a global id for that Resource. Other mentions found that doesn't
+      # match an existing Resource are returned as they are.
+      #
+      # @return [String] the content with the valid mentions replaced by a global id.
+      def rewrite
+        rewrited_content = parse_for_urls(content)
+        parse_for_ids(rewrited_content)
+      end
+
+      private
+
+      def parse_for_urls(content)
+        content.gsub(url_regex) do |match|
+          resource = resource_from_url_match(match)
+          if resource
+            update_metadata(resource)
+            resource.to_global_id
+          else
+            match
+          end
+        end
+      end
+
+      def parse_for_ids(content)
+        content.gsub(ID_REGEX) do |match|
+          resource = resource_from_id_match(Regexp.last_match(1))
+          if resource
+            update_metadata(resource)
+            resource.to_global_id
+          else
+            match
+          end
+        end
+      end
+
+      def resource_from_url_match(match)
```

Comment:
Shouldn't we also check here that that host of the URL matches with the organization host?

Reply:
@ahukkanen Should we also check is the host matches organziation.secondary_hosts?

Reply:
@roxanaopr The way you do it now is fine for me but could you check the latest comment I added below, please?